doc/conf.py

```diff
@@ -120,13 +120,35 @@ def __getattr__(cls, name):
     "sphinx.ext.napoleon",
     "sphinx.ext.todo",
     "sphinx.ext.viewcode",
+    "sphinxext.opengraph",
     "sphinx_automodapi.automodapi",
     "sphinx_automodapi.smart_resolver",
     "m2r2",
 ]
 
 intersphinx_mapping = {"https://docs.pennylane.ai/en/stable/": None}
 
+# OpenGraph Metadata
+ogp_title = "Catalyst Documentation"
+ogp_type = "website"
+ogp_description = (
+    "Catalyst is an experimental package that enables just-in-time (JIT) "
+    "compilation of PennyLane programs. Compile the entire "
+    "quantum-classical workflow."
+)
+ogp_use_first_image = True  # set to False for autocards
+ogp_image = "_static/catalyst_illustration.jpg"  # comment for autocards
+
+ogp_social_cards = {
+    "image": "_static/catalyst_illustration.jpg",
+    "enable": True,
+    "site_url": "https://docs.pennylane.ai/projects/catalyst",
+    "line_color": "#03b2ff",
+}
+
+# The base URL with a proper language and version.
+html_baseurl = os.environ.get("READTHEDOCS_CANONICAL_URL", "/")
```

Comment:
**Question:** What do we need this for?

Reply:
This is required so that the generated `og:image` URLs are correct. `READTHEDOCS_CANONICAL_URL` will have provide the base absolute URL docs.pennylane.ai, as well as the `/en/stable/` part.

It will also allow OG previews to work with RTDs PR previews :)